### DISCUSSION TITLE
New Demo: Working with Ports

### DISCUSSION THREAD

**Discussion by kumilingus** [clientIO/joint#1761] [News]
Are you **working with ports** and looking for inspiration? Check out our new [CodePen](https://codepen.io/jointjs/pen/rNdzYZj) showing the following features: 
- connection events
- connection validations
- magnet validations
- custom ports markups
- custom link tools
- delayed link tools removal
- highlight available ports

<img width="738" alt="image" src="https://user-images.githubusercontent.com/3967880/182562629-805f7ff0-e704-4f46-bc1b-d1306cb172f2.png">

**Comment by luisflavio** [comment]
Thank you so much for this brillant example.
